packages/headless/src/controllers/relevance-inspector/headless-relevance-inspector.ts

```diff
@@ -17,40 +30,150 @@ import {
   validateInitialState,
   validateOptions,
 } from '../../utils/validate-payload';
-import {buildController} from '../controller/headless-controller';
+import {buildController, Controller} from '../controller/headless-controller';
+
+export {
+  RankingInformation,
+  DocumentWeights,
+  TermWeightReport,
+  QueryRankingExpressionWeights,
+  ExecutionReport,
+  ExecutionStep,
+  UserIdentity,
+  RankingExpression,
+};
 
 export interface RelevanceInspectorProps {
+  /**
+   * The initial state that should be applied to the `RelevanceInspector` controller.
+   */
   initialState?: RelevanceInspectorInitialState;
+
+  /**
+   * The options for the `RelevanceInspector` controller.
+   */
   options?: RelevanceInspectorOptions;
 }
 
 const initialStateSchema = new Schema({
-  /**  If debug mode should be enabled */
   enabled: new BooleanValue({default: false}),
 });
 
 const optionsSchema = new Schema({
-  /** Whether to automatically log state to console on new responses */
   automaticallyLogInformation: new BooleanValue({default: true}),
 });
 
-export type RelevanceInspectorInitialState = SchemaValues<
-  typeof initialStateSchema
->;
-
-export type RelevanceInspectorOptions = SchemaValues<typeof optionsSchema>;
+export interface RelevanceInspectorInitialState {
+  /**
+   * Whether debug mode should be enabled.
+   * */
+  enabled?: boolean;
+}
 
-export type RelevanceInspectorState = RelevanceInspector['state'];
+export interface RelevanceInspectorOptions {
+  /**
+   * Whether to automatically log state to console on new responses.
+   * */
+  automaticallyLogInformation?: boolean;
+}
 
 /**
  * The `RelevanceInspector` controller is in charge of allowing displaying various debug information.
  */
-export type RelevanceInspector = ReturnType<typeof buildRelevanceInspector>;
+export interface RelevanceInspector extends Controller {
+  /**
+   * Enables debug mode.
+   */
+  enable(): void;
+
+  /**
+   * Disables debug mode.
+   */
+  disable(): void;
+
+  /**
+   * Logs debug information to the console.
+   */
+  logInformation(): void;
+
+  /**
+   * The state of the `RelevanceInspector` controller.
+   * */
+  state: RelevanceInspectorState;
+}
+
+export interface RelevanceInspectorState {
+  /**
+   * Whether debug mode is enabled.
+   * */
+  isEnabled: boolean;
+
+  /**
+   * The ranking information for every result.
+   */
+  rankingInformation?: ResultRankingInformation[];
+
+  /**
+   * The query execution report.
+   */
+  executionReport?: ExecutionReport;
 
+  /**
+   * The query expressions sent in the request.
+   */
+  expressions?: QueryExpressions;
+
+  /**
+   * The security identities.
+   */
+  userIdentities?: UserIdentity[];
+
+  /**
+   * The ranking expressions.
+   */
+  rankingExpressions?: RankingExpression[];
```

Comment:
There is an inconsistency in the docs and code, where some places use `RankingExpression`, others use `QueryRankingExpression`. I kept the existing interface name in this case, but we can change the name if you prefer.

Reply:
Yep, in our glossary it's 'query ranking expression', but we often refer to it as a 'ranking expression'. I think the inconsistency actually stems from the platform itself- the rules you can create are referred to as 'ranking expression rules'. Since we use both everywhere, I don't have a strong preference on which we use here.

Reply:
I have a very small preference for renaming it since I feel like that would make it easier to look up.

Reply:
Ok, I will rename the interface to `QueryRankingExpression`.